controllers/front/apiV2ShopHealthCheck.php

```diff
@@ -0,0 +1,222 @@
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/AFL-3.0 Academic Free License version 3.0
+ */
+
+use PrestaShop\Module\PsAccounts\Account\LinkShop;
+use PrestaShop\Module\PsAccounts\Account\Session\Firebase;
+use PrestaShop\Module\PsAccounts\Account\Session\ShopSession;
+use PrestaShop\Module\PsAccounts\Account\Token\NullToken;
+use PrestaShop\Module\PsAccounts\Account\Token\Token;
+use PrestaShop\Module\PsAccounts\Api\Client\AccountsClient;
+use PrestaShop\Module\PsAccounts\Api\Controller\AbstractShopRestController;
+use PrestaShop\Module\PsAccounts\Api\Controller\Request\ShopHealthCheckRequest;
+use PrestaShop\Module\PsAccounts\Provider\OAuth2\Oauth2Client;
+use PrestaShop\Module\PsAccounts\Provider\OAuth2\ShopProvider;
+use PrestaShop\Module\PsAccounts\Service\PsAccountsService;
+
+class ps_AccountsApiV2ShopHealthCheckModuleFrontController extends AbstractShopRestController
+{
+    /**
+     * @var LinkShop
+     */
+    private $linkShop;
+
+    /**
+     * @var Oauth2Client
+     */
+    private $oauth2Client;
+
+    /**
+     * @var ShopSession
+     */
+    private $shopSession;
+
+    /**
+     * @var Firebase\ShopSession
+     */
+    private $firebaseShopSession;
+
+    /**
+     * @var Firebase\OwnerSession
+     */
+    private $firebaseOwnerSession;
+
+    /**
+     * @var PsAccountsService
+     */
+    private $psAccountsService;
+
+    /**
+     * @var AccountsClient
+     */
+    private $accountsClient;
+
+    /**
+     * @var ShopProvider
+     */
+    private $shopProvider;
+
+    public function __construct()
+    {
+        parent::__construct();
+
+        // public healthcheck
+        $this->authenticated = false;
+
+        $this->linkShop = $this->module->getService(LinkShop::class);
+        $this->oauth2Client = $this->module->getService(Oauth2Client::class);
+        $this->shopSession = $this->module->getService(ShopSession::class);
+        $this->firebaseShopSession = $this->module->getService(Firebase\ShopSession::class);
+        $this->firebaseOwnerSession = $this->module->getService(Firebase\OwnerSession::class);
+        $this->accountsClient = $this->module->getService(AccountsClient::class);
+        $this->psAccountsService = $this->module->getService(PsAccountsService::class);
+        $this->shopProvider = $this->module->getService(ShopProvider::class);
+    }
+
+    /**
+     * ?fc=module&module=ps_accounts&controller=apiV1ShopHealthCheck&shop_id=1&autoheal
+     *
+     * @param Shop $shop
+     * @param ShopHealthCheckRequest $request
+     *
+     * @return array
+     *
+     * @throws Exception
+     */
+    public function show(Shop $shop, ShopHealthCheckRequest $request)
+    {
+        // refreshing one of firebase tokens will trigger a global refresh
+        $firebaseShopToken = $request->autoheal ?
```

Comment:
Why don't we just refresh token when it expires ? 
`GetOrRefreshToken`
The is no use of getting old token not refreshed to debug no?

Reply:
I suggest you add a comment on the EventModeling board